codegen/src/test/resources/software/amazon/awssdk/codegen/poet/client/test-batchmanager-async-class.java

```diff
@@ -0,0 +1,91 @@
+import software.amazon.awssdk.protocols.json.JsonOperationMetadata;
+import software.amazon.awssdk.services.batchmanager.model.BatchManagerException;
+import software.amazon.awssdk.services.sqs.batchmanager.SqsAsyncBatchManager;
+
+/**
+ * Internal implementation of {@link BatchManagerAsyncClient}.
+ *
+ * @see BatchManagerAsyncClient#builder()
+ */
+@Generated("software.amazon.awssdk:codegen")
+@SdkInternalApi
+final class DefaultBatchManagerAsyncClient implements BatchManagerAsyncClient {
+    private static final Logger log = LoggerFactory.getLogger(DefaultBatchManagerAsyncClient.class);
+
+    private final AsyncClientHandler clientHandler;
+
+    private final AwsJsonProtocolFactory protocolFactory;
+
+    private final SdkClientConfiguration clientConfiguration;
+
+    private final ScheduledExecutorService executorService;
+
+    protected DefaultBatchManagerAsyncClient(SdkClientConfiguration clientConfiguration) {
+        this.clientHandler = new AwsAsyncClientHandler(clientConfiguration);
+        this.clientConfiguration = clientConfiguration;
+        this.protocolFactory = init(AwsJsonProtocolFactory.builder()).build();
+        this.executorService = clientConfiguration.option(SdkClientOption.SCHEDULED_EXECUTOR_SERVICE);
+    }
+
+    @Override
+    public final String serviceName() {
+        return SERVICE_NAME;
+    }
+
+    @Override
+    public void close() {
+        clientHandler.close();
+    }
+
+    private <T extends BaseAwsJsonProtocolFactory.Builder<T>> T init(T builder) {
+        return builder.clientConfiguration(clientConfiguration).defaultServiceExceptionSupplier(BatchManagerException::builder)
+                      .protocol(AwsJsonProtocol.REST_JSON).protocolVersion("1.1");
+    }
+
+    private static List<MetricPublisher> resolveMetricPublishers(SdkClientConfiguration clientConfiguration,
+                                                                 RequestOverrideConfiguration requestOverrideConfiguration) {
+        List<MetricPublisher> publishers = null;
+        if (requestOverrideConfiguration != null) {
+            publishers = requestOverrideConfiguration.metricPublishers();
+        }
+        if (publishers == null || publishers.isEmpty()) {
+            publishers = clientConfiguration.option(SdkClientOption.METRIC_PUBLISHERS);
+        }
+        if (publishers == null) {
+            publishers = Collections.emptyList();
+        }
+        return publishers;
+    }
+
+    private HttpResponseHandler<AwsServiceException> createErrorResponseHandler(BaseAwsJsonProtocolFactory protocolFactory,
+                                                                                JsonOperationMetadata operationMetadata) {
+        return protocolFactory.createErrorResponseHandler(operationMetadata);
+    }
+
+    @Override
+    public SqsAsyncBatchManager batchManager() {
+        return SqsAsyncBatchManager.builder().client(this).scheduledExecutor(executorService).build();
```

Comment:
Can we use some other fake class here for testing? It is a bit confusing to use sqs batch manager here, to me

Reply:
Used another fake class for testing.